### PR TITLE
Fix search suggestions regression test

### LINKED ISSUE #15921: Changes to search query may be overridden by GlobalNavbar query update logic (racy)
The `search suggestions` regression test [is failing](https://github.com/sourcegraph/sourcegraph/issues/15178#issuecomment-729609488), with the search suggestions widget failing to show at [this step](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/web/src/regression/search.test.ts#L534:1).

Running the tests locally, I noticed that while the suggestions widget was initially shown, shortly after the [`jwtmi ` keyboard input](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@00c88f2e486dc5a9bb3aa4c42c85f399e9feac9c/-/blob/client/web/src/regression/search.test.ts#L531:1) was erased, and the suggestions widget was hidden.

I tracked this down to the [query update logic in `<GlobalNavbar/>`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@00c88f2e486dc5a9bb3aa4c42c85f399e9feac9c/-/blob/client/web/src/nav/GlobalNavbar.tsx#L118-141), which was added to fix another critical issue in #13954. Any query changes from user input occurring between a location change and the execution of this `useEffect` block would be erased.

### PR DESCRIPTION
**Pull request by lguychard**
Fixes #15921, #15178

My previous fix (#15922) was wrong: the issue here was simply caused by an unnecessary dependency on `location` in a `useEffect()` hook, causing its callback to be evaluated too often.

I validated this against the regression tests, but also went through with porting the search suggestions regression test to the integration test suite, where such tests should live, and adding an additional integration test validating the query update behaviour in `<GlobalNavbar/>`. I think this one is a winner :)

### CODE REVIEW

File: client/web/src/nav/GlobalNavbar.tsx [hunk 1]
```diff
@@ -138,7 +138,7 @@ export const GlobalNavbar: React.FunctionComponent<Props> = ({
         } else {
             onNavbarQueryChange({ query, cursorPosition: query.length })
         }
-    }, [interactiveSearchMode, isSearchRelatedPage, location, onFiltersInQueryChange, onNavbarQueryChange, query])
```
Comment (lguychard) [inline client/web/src/nav/GlobalNavbar.tsx:141]: This is the actual bugfix.

File: client/web/src/components/LinkWithIconOnlyTooltip.tsx [hunk 2]
```diff
@@ -18,7 +19,7 @@ export const LinkWithIconOnlyTooltip: React.FunctionComponent<{
     const LinkComponent = activeClassName ? NavLink : Link
     const linkProps = { to, className: `${className} d-flex align-items-center`, activeClassName }
     return (
-        <LinkComponent {...linkProps}>
+        <LinkComponent {...linkProps} data-testid={kebabCase(text)}>
```
Comment (lguychard) [inline client/web/src/components/LinkWithIconOnlyTooltip.tsx:22]: This is a change needed to add the "search field value" integration test.

File: client/web/src/integration/search.test.ts [hunk 3]
```diff
@@ -101,6 +102,104 @@ describe('Search', () => {
         await driver.page.click('.monaco-editor .view-lines')
     }
 
+    const getSearchFieldValue = (driver: Driver): Promise<string | undefined> =>
+        driver.page.evaluate(() => document.querySelector<HTMLTextAreaElement>('#monaco-query-input textarea')?.value)
+
+    describe('Suggestions', () => {
+        test('Typing in the search field shows relevant suggestions', async () => {
```
Comment (lguychard) [inline client/web/src/integration/search.test.ts:109]: This was directly ported from the regression test suite.

File: client/web/src/integration/search.test.ts [hunk 4]
```diff
@@ -101,6 +102,104 @@ describe('Search', () => {
...
+            expect(await getSearchFieldValue(driver)).toStrictEqual('foo')
+            // Field value is cleared when navigating to a non search-related page
+            await driver.page.waitForSelector('[data-testid="extensions"]')
+            await driver.page.click('[data-testid="extensions"]')
+            expect(await getSearchFieldValue(driver)).toStrictEqual('')
+            // Field value is restored when the back button is pressed
+            await driver.page.goBack()
+            expect(await getSearchFieldValue(driver)).toStrictEqual('foo')
+        })
+    })
```
Comment (lguychard) [inline client/web/src/integration/search.test.ts:201]: This is a new integration test validating the `GlobalNavbar` query update behaviour, previously untested.

**Review by eseliger — APPROVED** [review]
LGTM at quick glance but didn't get to test it manually because I'm in a call rn, but I assume you did :)

**Comment by lguychard** [comment]
Yup, I tested manually, and added an integration test validating what broke with the previous PR.

**Pull request merged**